### PR TITLE
Ensure that we cleanup whitespaces at ends

### PR DESCRIPTION
**Pull request by MihaiMadeT**

### CODE REVIEW

File: src/useCases/createUser.ts [hunk 1]
```diff
@@ -79,6 +79,16 @@ export default async (
   currentUser: Partial<User>,
   userDetails: Partial<User>
 ): PromiseResult<void> => {
+  const tmpUserDetails: Partial<User> = {}
```
Comment (emadkaramad) [inline src/useCases/createUser.ts:82]: I wonder if this logic should happen before validating fields on the page. Let's say that you insert white-space characters into a text box and then try saving the form. The value of the field is trimmed here and it uses an empty string in the query string.
What do you think?

Comment (MihaiMadeT) [reply src/useCases/createUser.ts:82]: I think we have discussed this offline a little bit before. If we were to throw a validation error, the user will be more annoyed than "Ah, sure I will remove the white space". I doubt there is a situation when adding a whitespace at the either end of the string would have been intended? They would most likely have made a mistake. Unless they are trying to input a password of course.

**Review by jaskaransarkaria — APPROVED** [review]
LGTM. I might have missed important bits of the conversation (probably have) but are we also ensuring that the data is correct and valid at point of input?

**Comment by MihaiMadeT** [comment]
> LGTM. I might have missed important bits of the conversation (probably have) but are we also ensuring that the data is correct and valid at point of input?

Yup, we do have checks in to check if data is valid before saving it. This is done in the 'const userFormIsValid' function

**Pull request merged**